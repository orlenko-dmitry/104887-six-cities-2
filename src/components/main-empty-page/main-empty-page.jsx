import React from 'react';
import {connect} from 'react-redux';
import {
  arrayOf,
  shape,
  number,
  string,
  func,
} from 'prop-types';

import MainTabs from '../main-tabs/main-tabs.jsx';
import actions from '../../store/actions.js';
import {getCities} from '../../store/selectors';

const MainEmptyPage = ({cities, city, selectCityHandler}) => (
  <div className="page page--gray page--main">
    <header className="header">
      <div className="container">
        <div className="header__wrapper">
          <div className="header__left">
            <a className="header__logo-link header__logo-link--active">
              <img
                className="header__logo"
                src="img/logo.svg"
                alt="6 cities logo"
                width={81}
                height={41}
              />
            </a>
          </div>
          <nav className="header__nav">
            <ul className="header__nav-list">
              <li className="header__nav-item user">
                <a
                  className="header__nav-link header__nav-link--profile"
                  href="#"
                >
                  <div className="header__avatar-wrapper user__avatar-wrapper"></div>
                  <span className="header__user-name user__name">
                    Oliver.conner@gmail.com
                  </span>
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <main className="page__main page__main--index page__main--index-empty">
      <h1 className="visually-hidden">Cities</h1>
      <MainTabs
        cities={cities}
        selectedCity={city}
        onSelectCityClick={selectCityHandler}
      />
      <div className="cities">
        <div className="cities__places-container cities__places-container--empty container">
          <section className="cities__no-places">
            <div className="cities__status-wrapper tabs__content">
              <b className="cities__status">No places to stay available</b>
              <p className="cities__status-description">
                We could not find any property availbale at the moment in {city.name}
              </p>
            </div>
          </section>
          <div className="cities__right-section" />
        </div>
      </div>
    </main>
  </div>
);

MainEmptyPage.propTypes = {
  city: shape({
    location: shape({
      latitude: number.isRequired,
      longitude: number.isRequired,
      zoom: number.isRequired,
    }).isRequired,
    name: string.isRequired,
  }).isRequired,
  cities: arrayOf(shape({})).isRequired,
  selectCityHandler: func.isRequired,
};

const mapStateToProps = (state) => ({
  city: state.city,
  cities: getCities(state),
});

const mapDispatchToProps = (dispatch) => ({
  selectCityHandler: (payload) => dispatch(actions.selectCity(payload)),
});

export {MainEmptyPage};

export default connect(mapStateToProps, mapDispatchToProps)(MainEmptyPage);
