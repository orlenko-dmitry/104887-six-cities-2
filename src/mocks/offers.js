const offers = [{
  "id": 1,
  "city": {
    "name": `Amsterdam`,
    "location": {
      "latitude": 52.370216,
      "longitude": 4.895168,
      "zoom": 10
    }
  },
  "previewImage": `img/apartment-01.jpg`,
  "images": [`/img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`],
  "title": `Beautiful & luxurious studio at great location`,
  "isFavorite": true,
  "isPremium": true,
  "rating": 4.8,
  "type": `apartment`,
  "bedrooms": 3,
  "max_adults": 4,
  "price": 120,
  "goods": [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  "host": {
    "id": 3,
    "isPro": true,
    "name": `Angelina`,
    "avatar_url": `img/avatar-angelina.jpg`
  },
  "description": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  "location": {
    "latitude": 52.35514938496378,
    "longitude": 4.673877537499948,
    "zoom": 8
  }
}, {
  "id": 2,
  "city": {
    "name": `Amsterdam`,
    "location": {
      "latitude": 52.370216,
      "longitude": 4.895168,
      "zoom": 10
    }
  },
  "previewImage": `img/room.jpg`,
  "images": [`/img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`],
  "title": `Wood and stone place`,
  "isFavorite": false,
  "isPremium": false,
  "rating": 4.2,
  "type": `private room`,
  "bedrooms": 1,
  "max_adults": 2,
  "price": 80,
  "goods": [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  "host": {
    "id": 3,
    "isPro": true,
    "name": `Angelina`,
    "avatar_url": `img/avatar-angelina.jpg`
  },
  "description": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  "location": {
    "latitude": 52.35514938496378,
    "longitude": 4.673877537499948,
    "zoom": 8
  }
}, {
  "id": 3,
  "city": {
    "name": `Amsterdam`,
    "location": {
      "latitude": 52.370216,
      "longitude": 4.895168,
      "zoom": 10
    }
  },
  "previewImage": `img/apartment-02.jpg`,
  "images": [`/img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`],
  "title": `Nice, cozy, warm big bed apartmen`,
  "isFavorite": true,
  "isPremium": false,
  "rating": 4,
  "type": `apartment`,
  "bedrooms": 2,
  "max_adults": 3,
  "price": 70,
  "goods": [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  "host": {
    "id": 3,
    "isPro": true,
    "name": `Angelina`,
    "avatar_url": `img/avatar-angelina.jpg`
  },
  "description": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  "location": {
    "latitude": 52.35514938496378,
    "longitude": 4.673877537499948,
    "zoom": 8
  }
}, {
  "id": 4,
  "city": {
    "name": `Amsterdam`,
    "location": {
      "latitude": 52.370216,
      "longitude": 4.895168,
      "zoom": 10
    }
  },
  "previewImage": `img/apartment-03.jpg`,
  "images": [`/img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`],
  "title": `Beautiful luxurious apartment at great location`,
  "isFavorite": false,
  "isPremium": true,
  "rating": 4.9,
  "type": `apartment`,
  "bedrooms": 3,
  "max_adults": 6,
  "price": 200,
  "goods": [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  "host": {
    "id": 3,
    "isPro": true,
    "name": `Angelina`,
    "avatar_url": `img/avatar-angelina.jpg`
  },
  "description": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  "location": {
    "latitude": 52.35514938496378,
    "longitude": 4.673877537499948,
    "zoom": 8
  }
}];

export default offers;
