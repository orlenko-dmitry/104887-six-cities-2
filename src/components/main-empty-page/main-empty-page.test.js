import React from 'react';
import {shallow} from 'enzyme';

import {MainEmptyPage} from './main-empty-page.jsx';
import offers from '../../mocks/offers.js';

const city = offers[0].city;

it(`MainPage renders correctly`, () => {
  const tree = shallow(
      <MainEmptyPage
        city={city}
        selectCityHandler={() => {}}
      />
  );

  expect(tree).toMatchSnapshot();
});
