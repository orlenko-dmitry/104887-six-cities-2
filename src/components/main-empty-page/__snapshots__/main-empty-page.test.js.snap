// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage renders correctly 1`] = `
<div
  className="page page--gray page--main"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height={41}
              src="img/logo.svg"
              width={81}
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="#"
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  Oliver.conner@gmail.com
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--index page__main--index-empty"
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <MainTabs
      cities={
        Array [
          Object {
            "location": Object {
              "latitude": 52.37454,
              "longitude": 4.897976,
              "zoom": 13,
            },
            "name": "Amsterdam",
          },
          Object {
            "location": Object {
              "latitude": 50.846557,
              "longitude": 4.351697,
              "zoom": 13,
            },
            "name": "Brussels",
          },
          Object {
            "location": Object {
              "latitude": 50.938361,
              "longitude": 6.959974,
              "zoom": 13,
            },
            "name": "Cologne",
          },
          Object {
            "location": Object {
              "latitude": 48.85661,
              "longitude": 2.351499,
              "zoom": 13,
            },
            "name": "Paris",
          },
          Object {
            "location": Object {
              "latitude": 53.550341,
              "longitude": 10.000654,
              "zoom": 13,
            },
            "name": "Hamburg",
          },
          Object {
            "location": Object {
              "latitude": 51.225402,
              "longitude": 6.776314,
              "zoom": 13,
            },
            "name": "Dusseldorf",
          },
        ]
      }
      onSelectCityClick={[Function]}
      selectedCity={
        Object {
          "location": Object {
            "latitude": 52.37454,
            "longitude": 4.897976,
            "zoom": 13,
          },
          "name": "Amsterdam",
        }
      }
    />
    <div
      className="cities"
    >
      <div
        className="cities__places-container cities__places-container--empty container"
      >
        <section
          className="cities__no-places"
        >
          <div
            className="cities__status-wrapper tabs__content"
          >
            <b
              className="cities__status"
            >
              No places to stay available
            </b>
            <p
              className="cities__status-description"
            >
              We could not find any property availbale at the moment in 
              Amsterdam
            </p>
          </div>
        </section>
        <div
          className="cities__right-section"
        />
      </div>
    </div>
  </main>
</div>
`;
